src/overlays/actors/ovl_Bg_Jya_Lift/z_bg_jya_lift.c

```diff
@@ -26,24 +36,128 @@ const ActorInit Bg_Jya_Lift_InitVars = {
     (ActorFunc)BgJyaLift_Destroy,
     (ActorFunc)BgJyaLift_Update,
     (ActorFunc)BgJyaLift_Draw,
+
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/func_80899BA0.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F4, 1400, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 1800, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 2500, ICHAIN_STOP),
+};
+extern s32      D_0600D7E8;
+extern Gfx      D_0600CCE0[];
+
+
+    void BgJyaLiftInitDynapoly(BgJyaLift* this, GlobalContext* globalCtx, u32 arg2, DynaPolyMoveFlag moveFlag) {
+    s32 pad1; 
+    u32 localConst;
+    localConst = 0;
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(arg2, (void*)&localConst);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, localConst);
+}
+
+void BgJyaLift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaLift* this = THIS;
+    this->unk_16A = 0;
+    if ((D_8089A020)) {
+        Actor_Kill(thisx);
+        return;
+    }
+    // Goddess lift CT
+    osSyncPrintf( "女神リフト CT\n"); 
+    BgJyaLiftInitDynapoly(this, globalCtx, &D_0600D7E8, 0);
+    Actor_ProcessInitChain(thisx, sInitChain);
+    if (Flags_GetSwitch(globalCtx, (thisx->params & 0x3F))) {
+        BgJyaLiftSetFinalPosY(thisx); 
+    } else {
+        BgJyaLiftSetInitPosY(thisx);
+    }
+    thisx->room = -1;
+    D_8089A020 = 1;
+    this->unk_16A = 1;
+}
+
+void BgJyaLift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaLift* this = THIS;
+    if (this->unk_16A != 0) {
+        // Goddess Lift DT
+        osSyncPrintf("女神リフト DT\n");
+        D_8089A020 = 0;
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+
+void BgJyaLiftSetInitPosY(BgJyaLift* this) {
+    this->actionFunc = func_80899D38;
+    this->dyna.actor.posRot.pos.y = 1613.0f;
+    this->unk_168 = 0;
+}
+
+
+void func_80899D38(BgJyaLift* this, GlobalContext* globalCtx) {
+    if ((Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F)) != 0) || (this->unk_168 > 0)) {
+        this->unk_168 = (this->unk_168 + 1);
+        if (this->unk_168 >= 0x14) {
+            func_800800F8(globalCtx, 0xD66, -0x63, &this->dyna.actor, 0);
+            func_80899DBC(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Lift/BgJyaLift_Init.s")
+void func_80899DBC(BgJyaLift* this) {
```

Comment:
BgJyaLift_SetupMove is a good name for this. Instead of a general "SetupAction" this is setting up a specific action